exercises/practice/protein-translation/src/lib.cairo

```diff
@@ -1,21 +1,15 @@
-#[derive(Destruct)]
-struct CodonsInfo {}
-
-pub fn parse(pairs: Array<(felt252, ByteArray)>) -> CodonsInfo {
-    // constructs a new CodonsInfo struct
-    panic!("implement `parse`")
+#[derive(Drop, Debug, PartialEq)]
+pub enum Proteins {
+    Methionine,
+    Phenylalanine,
+    Leucine,
+    Serine,
+    Tyrosine,
+    Cysteine,
+    Tryptophan,
+    STOP
```

Comment:
Just a thought, but do we need to include STOP here? It's not an actual protein, and it shouldn't be returned by `proteins` at any point.

Reply:
Makes sense. I [updated the terminology used for enums](https://github.com/exercism/cairo/pull/279/commits/bc0d9fc1137a703fab358cf06cfb870f95d06d0a):
- `enum Protein` is not `enum AminoAcid`, as this is the reality (proteins are chains of amino acids)
    - updated var names
- added a new `enum Codon` that has two variants: `AminoAcid: AminoAcid` (see above) and `Stop`; this way `Stop` is named as it should be and is never exported.